lib/puppet/configurer.rb

```diff
@@ -161,6 +161,7 @@ def run_internal(options)
         begin
           if node = Puppet::Node.indirection.find(Puppet[:node_name_value],
               :environment => Puppet::Node::Environment.remote(@environment),
+              :configured_environment => Puppet[:environment],
```

Comment:
IIUC, `environment` and `configured_environment` are always equal at this point? I say because the Configurer initializer sets `@environment = Puppet[:environment]`. That said, I think it's good to be explicit and always send the param when making the node request.